src/middlewared/middlewared/plugins/jail.py

```diff
@@ -42,8 +44,27 @@ def query(self, filters=None, options=None):
         try:
             jail_dicts = ioc.IOCage(
                 jail=jail_identifier).get('all', recursive=recursive)
-            jails = [list(jail.values())[0] for jail in jail_dicts
-                     ] if jail_identifier is None else [jail_dicts]
+            for jail in jail_dicts:
+                jail = list(jail.values())[0]
+                if jail['dhcp'] == 'on':
+                    interface = jail["interfaces"].split(',')[0].split(
+                        ':')[0]
+                    uuid = jail['host_hostuuid']
+
+                    if interface == "vnet0":
+                        # Inside jails they are epairNb
+                        interface = \
+                            f"{interface.replace('vnet', 'epair')}b"
+
+                    if jail['state'] == 'up':
```

Comment:
Seems like all the interface checks above should be done here, no point in making them if jail is not up